linq4j/src/main/java/org/apache/calcite/linq4j/tree/ConstantExpression.java

```diff
@@ -245,6 +249,31 @@ private static ExpressionWriter map(ExpressionWriter writer, Map map,
     return writer.append(end);
   }
 
+  private static ExpressionWriter writeSet(ExpressionWriter writer, Set set) {
+    writer.append("com.google.common.collect.ImmutableSet.");
+    if (set.isEmpty()) {
+      return writer.append("of()");
+    }
+    if (set.size() < 5) {
+      return set(writer, set, "of(", ",", ")");
+    }
+    return set(writer, set, "builder().add(", ")\n.add(", ").build()");
```

Comment:
Can we use this method?
```
public Builder<E> add(E... elements)
```
So that L257~L260 can be merged.
And the `set` method is not needed, you can replace it with `String.join` or `Stream` with `StringJoiner`.

Reply:
Thanks for you review.Maybe we should follow the the 'map()' method.